KOVETO.CPP

```diff
@@ -62,39 +69,51 @@ static int grass_line_heightmap(gyuru* poly, int v1, int v2, int* x0, int cur, i
         heightmap[0] = y1;
     }
 
+    // Skip lines of length 0.
     if (x1 >= x2) {
         return cur;
     }
+
+    // No more room in the heightmap.
     if (x1 - *x0 >= max_heightmap_length) {
         return cur;
     }
 
+    // Skip if we've somehow jumped forward to the right.
     if (cur < x1 - 1) {
 #ifdef DEBUG
         hiba("o96ytuyg");
 #endif
         return cur;
     }
 
+    // Calculate the slope of the line between the two
+    // vertices, the y value at each step is recorded in the
+    // heightmap.
     for (int x = x1; x <= x2; x++) {
+        // We've doubled back to the left on a previous line, don't overwrite
+        // existing line data.
         if (x < cur) {
             continue;
         }
+
+        // We've reached the end of the heightmap.
         if (x - *x0 >= max_heightmap_length) {
             return cur;
         }
+
+        // y = y1 + (dy/dx) * (x1 - x2)
```

Comment:
Strange comment

Reply:
What I mean is can the code just be rewritten to be clearer instead of a comment?

Reply:
Will have a look, I did think about trying to write a bunch here, about linear interpolation and stuff tooo..